From 0445980feb3099945674c28da3bd8b329a4e2ff1 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@collabora.com>
Date: Fri, 15 Mar 2019 14:39:40 +0000
Subject: [PATCH 2/2] vendor/getinfo: Don't assume that Ubuntu is the last
 vendor

If your dpkg-vendor configuration in base-files is an Ubuntu derivative,
but not exactly Ubuntu, and not an exact match for anything in vendor/,
we want to use vendor/ubuntu.

Previously we used whatever $DISTRO happened to be at the end of the
"for" loop above, but that's undesirable: even if we assume that find(1)
returns results in alphabetical order (which is not guaranteed), we
wouldn't want to return a non-Ubuntu result if a future apt version
happens to have a vendor that sorts later than vendor/ubuntu, for
example vendor/zombieos or something.

Closes: #924662
Signed-off-by: Simon McVittie <smcv@collabora.com>
---
 vendor/getinfo | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/vendor/getinfo b/vendor/getinfo
index 8830c238f..2ef257c90 100755
--- a/vendor/getinfo
+++ b/vendor/getinfo
@@ -28,7 +28,7 @@ getcurrent() {
 
 	# Do the ubuntu/debian dance we talked about
 	if dpkg-vendor --derives-from ubuntu; then
-		echo $DISTRO
+		echo ubuntu
 		return 0
 	fi
 
-- 
2.20.1

